test/plugin/base_test.rb

```diff
@@ -320,6 +324,30 @@ def setup
     'cacheValidatedWithOriginServer' => true
   }
 
+  # Tags and their sanitized and encoded version.
+  VALID_TAGS = {
+    'test' => 'test',
+    'germanß' => 'german%C3%9F',
+    'chinese中' => 'chinese%E4%B8%AD',
+    'specialCharacter/_-.' => 'specialCharacter%2F_-.',
+    'abc@&^$*' => 'abc%40%26%5E%24%2A',
+    '@&^$*' => '%40%26%5E%24%2A'
+  }
+  NON_STRING_TAGS = {
```

Comment:
You almost always join these three into one hash. How about defining only one `INVALID_TAGS` hash, and then in the one case where you only need nonempty string tags, filtering for those:
```
INVALID_TAGS.select { |k, _| k.is_a? String and !k.empty?}
```